### PR TITLE
Issue project 192

### PR DESCRIPTION
**Pull request by jbaublitz**
Related to stratis-storage/project#192
Supersedes #2253

### CODE REVIEW

File: src/engine/strat_engine/backstore/blockdevmgr.rs [hunk 1]
```diff
@@ -466,11 +464,22 @@ impl BlockDevMgr {
...
     pub fn key_desc(&self) -> Option<&KeyDescription> {
-        self.key_desc.as_ref()
+        let mut iter = self
+            .block_devs
+            .iter()
+            .filter_map(|bd| bd.key_description())
+            .unique();
+        let key_desc = iter.next();
+
+        // Liminal device code will note set up a pool with multiple key description
```
Comment (mulkieran) [inline src/engine/strat_engine/backstore/blockdevmgr.rs:474]: ^note^not

**Review by mulkieran — CHANGES_REQUESTED** [review]
typo...

File: src/engine/strat_engine/backstore/blockdevmgr.rs (hunk 1 above)
Comment (jbaublitz) [reply src/engine/strat_engine/backstore/blockdevmgr.rs]: Resolved.

**Comment by jbaublitz** [comment]
- [ ] Take key_desc out of BlockDevMgr struct
- [ ] Stow Clevis config information in every StratBlockDev
- [ ] Handle setting of the Clevis config information in the StratBlockDev after a bind operation
- [ ] Export the Clevis config for a pool via the FetchProperties interface over the D-Bus
- [ ] Handle getting the Clevis config during an add operation

**Comment by jbaublitz** [comment]
@mulkieran I'm leaving the method `CryptHandle::clevis_info()` and allowing the dead code as you will need that for your work in the liminal devices code.

**Comment by jbaublitz** [comment]
I'm going to do one more pass but I've tested the bind command, the report generated for both the sim and strat engine, and everything seems to be working (including unlock; @drckeefe, I was unable to reproduce your issue with real devices, but please let me know if you have a way to).

@mulkieran If you want to work on the part of this, please go ahead and I'll put mine in review once it's ready.

**Pull request closed without merging**